client/fs_endpoint_test.go

```diff
@@ -74,48 +74,31 @@ func TestFS_Stat_NoAlloc(t *testing.T) {
 }
 
 func TestFS_Stat(t *testing.T) {
-	t.Skip("missing mock driver plugin implementation")
 	t.Parallel()
 	require := require.New(t)
 
-	// Start a client
-	c, cleanup := TestClient(t, nil)
+	// Start a server and client
+	s := nomad.TestServer(t, nil)
+	defer s.Shutdown()
+	testutil.WaitForLeader(t, s.RPC)
+
+	c, cleanup := TestClient(t, func(c *config.Config) {
+		c.Servers = []string{s.GetConfig().RPCAddr.String()}
+	})
 	defer cleanup()
 
 	// Create and add an alloc
-	a := mock.Alloc()
-	task := a.Job.TaskGroups[0].Tasks[0]
-	task.Driver = "mock_driver"
-	task.Config = map[string]interface{}{
+	job := mock.BatchJob()
+	job.TaskGroups[0].Count = 1
+	job.TaskGroups[0].Tasks[0].Config = map[string]interface{}{
 		"run_for": "500ms",
 	}
-	c.addAlloc(a, "")
-
-	// Wait for the client to start it
-	testutil.WaitForResult(func() (bool, error) {
-		ar, ok := c.allocs[a.ID]
-		if !ok {
-			return false, fmt.Errorf("alloc doesn't exist")
-		}
-
-		alloc := ar.Alloc()
-		running := false
-		for _, s := range alloc.TaskStates {
-			if s.State == structs.TaskStateRunning {
-				running = true
-			} else {
-				running = false
-			}
-		}
-
-		return running, fmt.Errorf("tasks not running")
-	}, func(err error) {
-		t.Fatal(err)
-	})
+	// Wait for alloc to be running
+	alloc := testutil.WaitForRunning(t, s.RPC, job)[0]
```

Comment:
Phooey, just noticed this is racy. In tests like this we don't actually care if the alloc is _currently_ running. The test will work even if the batch job finishes and the alloc is terminal. `WaitForRunning` asserts _currently_ running only, so this could fail on slow test runners like CI or VMs.

Some solutions:
* Bump `run_for` to "10s" or something and just let the deferred Client cleanup kill it (the client should be in dev mode so the cleanup will destroy running allocs)
* Change WaitForRunning although this may break other tests that want this exact behavior.
* Use a service job and just let it run forever (until Client cleanup)

Reply:
I went for the second option for having WaitForRunning checking if tasks are out in pending state.  There are few tests that call it (mostly the functions here), and callers need to handle case where allocs terminated right after WaitForRunning finished anyway.